pkg/clients/network/network.go

```diff
@@ -182,7 +182,7 @@ func GeneratePublicIPAddressObservation(az networkmgmt.PublicIPAddress) *v1alpha
 	v.ID = azure.ToString(az.ID)
 	v.Address = azure.ToString(az.IPAddress)
 	v.Version = string(az.PublicIPAddressVersion)
-	if az.IPConfiguration != nil {
+	if az.PublicIPAddressPropertiesFormat != nil && az.PublicIPAddressPropertiesFormat.IPConfiguration != nil && az.PublicIPAddressPropertiesFormat.IPConfiguration.IPConfigurationPropertiesFormat != nil {
 		v.IPConfiguration = &v1alpha3.IPConfiguration{
 			PrivateIPAllocationMethod: string(az.IPConfiguration.PrivateIPAllocationMethod),
```

Comment:
Are all the referenced fields (`PrivateIPAllocationMethod`, `PrivateIPAddress` and `ProvisioningState`) embedded by the same struct (`IPConfigurationPropertiesFormat`)?

Reply:
Hi @ulucinar I checked, all the referenced fields are embedded by the same struct.